controlplane/kubeadm/controllers/controller.go

```diff
@@ -554,7 +554,7 @@ func (r *KubeadmControlPlaneReconciler) adoptOwnedSecrets(ctx context.Context, k
 
 	for i := range secrets.Items {
 		s := secrets.Items[i]
-		if !util.IsControlledBy(&s, currentOwner) {
+		if !util.PointsTo(s.GetOwnerReferences(), currentOwner) {
```

Comment:
To avoid hitting an issue if UIDs may not match due to resources being moved to a different cluster.
```suggestion
		if !util.IsOwnedByObject(s.GetOwnerReferences(), currentOwner) {
```